navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java

```diff
@@ -24,19 +25,22 @@
   @Override
   public boolean isUserOffRoute(Location location, RouteProgress routeProgress, MapboxNavigationOptions options) {
     Position futurePosition = getFuturePosition(location, options);
-    double radius = Math.min(options.maximumDistanceOffRoute(),
+    double radius = Math.max(options.maximumDistanceOffRoute(),
       location.getAccuracy() + options.userLocationSnapDistance());
 
-    boolean isOffRoute = userTrueDistanceFromRoute(futurePosition, routeProgress) > radius;
+    LegStep currentStep = routeProgress.currentLegProgress().currentStep();
+    boolean isOffRoute = userTrueDistanceFromStep(futurePosition, currentStep) > radius;
 
     // If the user is moving away from the maneuver location and they are close to the next step we can safely say they
     // have completed the maneuver. This is intended to be a fallback case when we do find that the users course matches
     // the exit bearing.
     boolean isCloseToUpcomingStep;
 
-    if (routeProgress.currentLegProgress().upComingStep() != null) {
-      isCloseToUpcomingStep = userTrueDistanceFromRoute(futurePosition, routeProgress) < radius;
+    LegStep upComingStep = routeProgress.currentLegProgress().upComingStep();
+    if (upComingStep != null) {
+      isCloseToUpcomingStep = userTrueDistanceFromStep(futurePosition, upComingStep) < radius;
       if (isOffRoute && isCloseToUpcomingStep) {
+        // TODO increment step index
```

Comment:
Should be able to pass the `indices` reference and use `NavigationHelper.increaseIndices()` to increase.

Reply:
Sweet - @ericrwolfe do you want to do this now? Or should I ticket

Reply:
Be mindful that the name of this method `isUserOffRoute()` implies it shouldn't have any side effects, particularly if this were to be called somewhere else.

Probably best to move the `isCloseToUpcomingStep` check and the `increaseIndices()` call out into `handleRequest` like we talked about.